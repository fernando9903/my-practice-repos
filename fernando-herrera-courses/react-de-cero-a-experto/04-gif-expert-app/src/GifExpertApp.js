import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ({initialState=[]}) => {

    const [categories, setCategories] = useState(initialState)

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory
                setCategories={setCategories}
            />
            <hr />
            {}
            <ol>
                {
                    categories.map( (category) =>
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    )
                }
            </ol>
        </>
    );
}

export default GifExpertApp;