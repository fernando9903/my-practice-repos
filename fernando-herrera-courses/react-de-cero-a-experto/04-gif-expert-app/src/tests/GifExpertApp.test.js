import React from "react";
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GifExpertApp from "../GifExpertApp";

describe('Test on the <GifExpertApp> component', () => {
    test('The component is renderign correctly', () => {
        const wraper = shallow(
            <GifExpertApp/>
        );
        expect(wraper).toMatchSnapshot();
    })
    test('Debe de mostrar una lista de categorias', () => {
        const categories =  ['One push', 'Dragon ball'];
        const wraper = shallow(
            <GifExpertApp
                initialState={categories}
            />
        );
        expect(wraper).toMatchSnapshot();
        expect(wraper.find('GifGrid').length).toBe(categories.length);
    })
})