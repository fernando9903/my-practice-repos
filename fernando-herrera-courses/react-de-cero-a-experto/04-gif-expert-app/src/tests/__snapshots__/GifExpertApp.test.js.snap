// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the <GifExpertApp> component Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One push"
      key="One push"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Test on the <GifExpertApp> component The component is renderign correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
